src/main/java/it/trade/api/TradeItApiClient.java

```diff
@@ -96,231 +75,164 @@ private void injectSession(TradeItRequestWithSession request) {
+        this.statelessTradeItApiClient.getOAuthLoginPopupUrlForTokenUpdate(
+                this.apiKey,
+                broker,
+                userId,
+                userToken,
+                deepLinkCallback,
+                callback
+        );
     }
 
     public void linkBrokerWithOauthVerifier(String oAuthVerifier, final TradeItCallback<TradeItLinkedLogin> callback) {
-        final TradeItOAuthAccessTokenRequest request = new TradeItOAuthAccessTokenRequest(this.apiKey, oAuthVerifier);
-        tradeItApi.getOAuthAccessToken(request).enqueue(new DefaultCallbackWithErrorHandling<TradeItOAuthAccessTokenResponse, TradeItLinkedLogin>(callback) {
-            @Override
-            public void onSuccessResponse(Response<TradeItOAuthAccessTokenResponse> response) {
-                TradeItLinkedLogin linkedLogin = new TradeItLinkedLogin(request, response.body());
-                callback.onSuccess(linkedLogin);
-            }
-        });
+        this.statelessTradeItApiClient.linkBrokerWithOauthVerifier(this.apiKey, oAuthVerifier, callback);
     }
 
     public void unlinkBrokerAccount(TradeItLinkedLogin linkedLogin, final TradeItCallback callback) {
-        TradeItUnlinkLoginRequest request = new TradeItUnlinkLoginRequest(this.apiKey, linkedLogin);
-        tradeItApi.unlinkLogin(request).enqueue(new DefaultCallbackWithErrorHandling<TradeItResponse, TradeItResponse>(callback) {
-            @Override
-            public void onSuccessResponse(Response<TradeItResponse> response) {
-                callback.onSuccess(response.body());
-            }
-        });
+        this.statelessTradeItApiClient.unlinkBrokerAccount(this.apiKey, linkedLogin, callback);
     }
 
-    public void authenticate(TradeItLinkedLogin linkedLogin, final AuthenticationCallback<TradeItAuthenticateResponse, ?> callback) {
+    public void authenticate(
+            TradeItLinkedLogin linkedLogin,
+            final AuthenticationCallback<TradeItAuthenticateResponse, ?> callback
+    ) {
         if (serverUuid == null) {
             serverUuid = UUID.randomUUID().toString();
         }
 
         TradeItAuthenticateRequest authenticateRequest = new TradeItAuthenticateRequest(this.apiKey, linkedLogin);
-        authenticateRequest.serverUuid = serverUuid;
-
-        tradeItApi.authenticate(authenticateRequest).enqueue(new Callback<TradeItAuthenticateResponse>() {
-            public void onResponse(Call<TradeItAuthenticateResponse> call, Response<TradeItAuthenticateResponse> response) {
-                if (response.isSuccessful()) {
-                    TradeItAuthenticateResponse authenticateResponse = response.body();
-                    if (authenticateResponse.status == TradeItResponseStatus.SUCCESS || authenticateResponse.status == TradeItResponseStatus.INFORMATION_NEEDED) {
-                        sessionToken = authenticateResponse.sessionToken;
+        this.statelessTradeItApiClient.tradeItApi.authenticate(authenticateRequest).enqueue(
```

Comment:
Why can't this just be a passthrough call to the stateless client like the other calls? Why do we need to use the internal `tradeItApi`?

Reply:
It is because we need to set the `sessionToken` variable of the `TradeItApiClient`:

```java
sessionToken = authenticateResponse.sessionToken
```